### PR TITLE
fix(install): verify addon Give dependency before update #3217

### LINKED ISSUE #3217: feat(admin): add notice to add-ons warning to always check if Give Core is up-to-date.
## User Story

As a `site administrator`, I want `to be aware that Give core needs to be updated before any add-ons` so that `after performing updates all of my add-ons remain active`.

Every major release, the support team gets a handful of tickets from customers panicking because their forms are broken, when in reality it is simply that they updated add-ons before updating core, and the add-ons deactivate themselves.

## Current Behavior
I currently click through and update the plugins, and if I go alphabetically and I am using the Give Authorize.net add-on, it updates before the Give Donation plugin and deactivates itself if it doesn't pass the minimum give version test.

## Expected Behavior

I expect to be warned if a new update to Give core is necessary before updating an add-on.

## Possible Solution

Since the minimum give version check technically happens in the add-on code, this might require issues in each add-on repo, but I was thinking about a way to add aWarning any time Give Core is not on the latest stable version to the top of the plugins page, or to the top of each add-on's listing (search for `give-` in the list?) warning to update core first.

## Steps to Reproduce
1. Downgrade Give to 1.8.x
2. Try to update the recurring add-on to the latest.
3. It'll deactivate.

## Tasks

- [ ] warn user they are about to update when Give is not on the latest version.

### PR DESCRIPTION
**Pull request by ravinderk**
## Description
This pr will resolve #3217

## How Has This Been Tested?
I tested this pr by updating multiple licensed Give addon.
Testing video: https://drive.google.com/file/d/1Jrdy9wQdiQirzid0CncdGmO9RyiSoECL/view?
usp=sharing
Snippet link: https://givewp.slack.com/files/U1DDJMUTV/FAWLZ3GSD/give_pull_3268.php
## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/actions.php [hunk 1]
```diff
@@ -331,4 +331,61 @@ function give_update_log_form_id( $args ) {
...
+
+	$parser           = new Give_Readme_Parser( $url );
+	$give_min_version = $parser->requires_at_least();
+
+
+	if ( version_compare( GIVE_VERSION, $give_min_version, '<' ) ) {
+		return new WP_Error(
+			'Give_Addon_Update_Error',
+			sprintf(
+				__( 'You need Give version %s to update this plugin.', 'give' ),
```
Comment (ravinderk) [inline includes/actions.php]: @DevinWalker Need you help to update this text

File: includes/class-give-readme-parser.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	private $file_content;
+
+	/**
+	 * Give_Readme_Parser constructor.
+	 *
+	 * @param string $file_url
+	 */
+	function __construct( $file_url ) {
+		$this->file_url = $file_url;
+		$this->file_content = ! empty( $this->file_url ) ? file_get_contents( $this->file_url ) : '';
```
Comment (raftaar1191) [inline includes/class-give-readme-parser.php]: @ravinderk 

In WordPress, we have a specific function to retrieve the content from URL 

We should used that in place of Core PHP functions

**Comment by raftaar1191** [comment]
@ravinderk I tested this PR by updating the Give Add-on plus non-Give Plugin to. It's working as expected and awesome 


image

File: includes/actions.php (hunk 1 above)
Comment (DevinWalker) [reply includes/actions.php]: @ravinderk 

> Give version %s is required to update this add-on.

File: includes/class-give-readme-parser.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	 * Get required Give core minimum version for addon
+	 *
+	 * @since 2.1.4
+	 * @access public
+	 *
+	 * @return string
+	 */
+	public function requires_at_least() {
+		// Regex to extract Give core minimum version from the readme.txt file.
+		preg_match('|Give requires at least:(.*)|i', $this->file_content, $_requires_at_least );
```
Comment (DevinWalker) [inline includes/class-give-readme-parser.php]: Let's parse this text instead:

> Requires Give: 2.1

**Comment by ravinderk** [comment]
@DevinWalker Changes done.

**Pull request merged**